### PR TITLE
CC-16979: Group refund items costs with items by charge-id

### PR DESCRIPTION
**Pull request by michbeck**
Developer(s): @michbeck

Ticket: https://spryker.atlassian.net/browse/CC-16934

### CODE REVIEW

**Review by AsonUnique — APPROVED** [review]

**Review by romanhavrylko — APPROVED** [review]

File: src/SprykerEco/Zed/Unzer/Business/Refund/UnzerRefundExpander.php [hunk 1]
```diff
@@ -147,27 +147,36 @@ protected function getChildUnzerCredentialsCollection(UnzerCredentialsTransfer $
...
+
+        return $refundTransfer;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\ExpenseTransfer $expenseTransfer
+     *
+     * @return \Generated\Shared\Transfer\UnzerRefundItemTransfer
+     */
+    protected function createUnzerRefundItemTransfer(ExpenseTransfer $expenseTransfer): UnzerRefundItemTransfer
```
Comment (kkicheglovspryker) [inline src/SprykerEco/Zed/Unzer/Business/Refund/UnzerRefundExpander.php]:
```suggestion
    protected function createExpenseUnzerRefundItemTransfer(ExpenseTransfer $expenseTransfer): UnzerRefundItemTransfer
```

File: src/SprykerEco/Zed/Unzer/Business/Refund/UnzerRefundExpander.php [hunk 2]
```diff
@@ -147,27 +147,36 @@ protected function getChildUnzerCredentialsCollection(UnzerCredentialsTransfer $
...
+    {
+        return (new UnzerRefundItemTransfer())->setBasketItemReferenceId(
+            sprintf(UnzerConstants::UNZER_BASKET_SHIPMENT_REFERENCE_ID_TEMPLATE, $expenseTransfer->getUnzerParticipantIdOrFail()),
+        )
+            ->setQuantity(UnzerConstants::PARTIAL_REFUND_QUANTITY)
```
Comment (kkicheglovspryker) [inline src/SprykerEco/Zed/Unzer/Business/Refund/UnzerRefundExpander.php]: As soon as expenses will be refunded only fully - we can provide new const - UnzerConstants::FULL_REFUND_QUANTITY with value 1

File: src/SprykerEco/Zed/Unzer/Business/Refund/UnzerRefundExpander.php (hunk 1 above)
Comment (michbeck) [reply src/SprykerEco/Zed/Unzer/Business/Refund/UnzerRefundExpander.php]: Doesn't it make sense to name the method in such a way that it says what is being created?

Comment (michbeck) [reply src/SprykerEco/Zed/Unzer/Business/Refund/UnzerRefundExpander.php]: Fixed.

File: src/SprykerEco/Zed/Unzer/Business/Refund/UnzerRefundExpander.php (hunk 2 above)
Comment (michbeck) [reply src/SprykerEco/Zed/Unzer/Business/Refund/UnzerRefundExpander.php]: Fixed.

**Review by kkicheglovspryker — APPROVED** [review]

**Review by DmitryLymarenko — APPROVED** [review]

**Review by kkicheglovspryker — APPROVED** [review]

**Review by kkicheglovspryker — APPROVED** [review]

**Pull request merged**